spring-batch-infrastructure/src/test/java/org/springframework/batch/item/json/builder/JsonItemReaderBuilderTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.batch.item.json.builder;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import org.springframework.batch.item.json.JsonItemReader;
+import org.springframework.batch.item.json.JsonObjectReader;
+import org.springframework.core.io.Resource;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.springframework.test.util.ReflectionTestUtils.getField;
+
+/**
+ * @author Mahmoud Ben Hassine
+ */
+public class JsonItemReaderBuilderTest {
+
+	@Mock
+	private Resource resource;
+	@Mock
+	private JsonObjectReader<String> jsonObjectReader;
+
+	@Before
+	public void setUp() {
+		MockitoAnnotations.initMocks(this);
+	}
+
+	@Test
+	public void testValidation() {
+		try {
+			new JsonItemReaderBuilder<String>()
+					.build();
+			fail("A json object reader is required.");
+		}
+		catch (IllegalArgumentException iae) {
+			assertEquals("A json object reader is required.",
+					iae.getMessage());
+		}
+
+		try {
+			new JsonItemReaderBuilder<String>()
+					.jsonObjectReader(this.jsonObjectReader)
+					.build();
+			fail("A resource is required.");
+		}
+		catch (IllegalArgumentException iae) {
+			assertEquals("A resource is required.",
+					iae.getMessage());
+		}
+
+		try {
+			new JsonItemReaderBuilder<String>()
+					.jsonObjectReader(this.jsonObjectReader)
+					.resource(this.resource)
+					.saveState(true)
```

Comment:
This line is redundant isn't it (save state defaults to `true`)?

Reply:
I thought being explicit is better for this specific test case. But I will remove it, I agree it is not really necessary.

Reply:
fixed in 5b6e7b9ff7db438e5ef574b4d9fc1ea4b5aa5d9f